File: EnnuiBackend/src/main/java/at/ennui/backend/games/model/GameDto.java
package at.ennui.backend.games.model;

import at.ennui.backend.games.configuration.GameCategories;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Martin Singer on 19.09.2017.
 */
public class GameDto {
    private Long id;
    private String name;
    private String description;
    private String instruction;
    private String cover;
    private int minPlayer;
    private int maxPlayer;
    private boolean activated;
    private long ownerId;
    private List<GameCategories> categories;
    private int rating;
    private boolean ratedByUser;

    public boolean isRatedByUser() {
        return ratedByUser;
    }

    public void setRatedByUser(boolean ratedByUser) {
        this.ratedByUser = ratedByUser;
    }

    public int getRating() {
        return rating;
    }

    public void setRating(int rating) {
        this.rating = rating;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getInstruction() {
        return instruction;
    }

    public void setInstruction(String instruction) {
        this.instruction = instruction;
    }

    public String getCover() {
        return cover;
    }

    public void setCover(String cover) {
        this.cover = cover;
    }

    public int getMinPlayer() {
        return minPlayer;
    }

    public void setMinPlayer(int minPlayer) {
        this.minPlayer = minPlayer;
    }

    public int getMaxPlayer() {
        return maxPlayer;
    }

    public void setMaxPlayer(int maxPlayer) {
        this.maxPlayer = maxPlayer;
    }

    public boolean isActivated() {
        return activated;
    }

    public void setActivated(boolean activated) {
        this.activated = activated;
    }

    public long getOwnerId() {
        return ownerId;
    }

    public void setOwnerId(long ownerId) {
        this.ownerId = ownerId;
    }

    public void addCategory(GameCategories categories){
        if(getCategories() == null){
            setCategories(new ArrayList<>());
        }
        this.categories.add(categories);
    }

    public List<GameCategories> getCategories() {
        return categories;
    }

    public void setCategories(List<GameCategories> categories) {
        this.categories = categories;
    }
}

File: EnnuiFrontend/app/scripts/DTOs/GameDTO.ts
export class GameDTO{
    id:number;
    name:string
    description:string;
    instruction:string;
    cover:string;
    minPlayer:number;
    maxPlayer:number;
    categories: Array<String>;
    rating:number;
    ratedByUser: boolean;
    activated: boolean;
    ownerId: number;

    constructor(_id:number,_name:string,_description:string,_instruction:string,_cover:string,_minPlayer:number,_maxPlayer:number,_category:Array<String>,_rating:number, ratedByUser: boolean, activated: boolean, ownerId:number){
        this.id = _id;
        this.name = _name;
        this.description = _description;
        this.instruction = _instruction;
        this.cover = _cover;
        this.minPlayer = _minPlayer;
        this.maxPlayer = _maxPlayer;
        this.categories = _category;
        this.rating = _rating;
        this.ratedByUser = ratedByUser;
        this.activated = activated;
        this.ownerId = ownerId;
    }

    public getActivated(){
        return this.activated;
    }

    public getRatedByUser(){
        return this.ratedByUser;
    }
    
    public getId(){
        return this.id;
    }

    public getName(){
        return this.name;
    }
    
    public getDescription(){
        return this.description;
    }

    public getInstruction(){
        return this.instruction;
    }

    public getCover(){
        return this.cover;
    }

    public getMinPlayer(){
        return this.minPlayer;
    }

    public getMaxPlayer(){
        return this.maxPlayer;
    }

    public getCategories(){
        return this.categories;
    }

    public getRating(){
        return this.rating;
    }
}